### DISCUSSION TITLE
3.0能否加上动画过渡和定位组件

### DISCUSSION THREAD

**Discussion by loyaoo** [layui/layui#1779] [讨论]
### 议题条件

- [X] 我确认已查看官方使用文档：https://layui.dev ，但没有找到相关解决方案。
- [X] 我确认已在 [Issues](https://github.com/layui/layui/issues?q=is%3Aissue ) 中搜索过类似的问题，但没有找到相关解决方案。
- [X] 我已仔细阅读： 🍀[ Layui Issue 贡献指南](https://github.com/layui/layui/blob/main/CONTRIBUTING.md)

### 议题类型

功能请求

### 使用版本

v2.9.8

### 问题描述

2.8的下拉组件没有动画效果和实时定位效果，vue的<transition>组件非常好用enter-from，enter-active，enter-to；搭配popperJS定位组件，可以完美解决弹出层的精美动画效果；有了popper，系统内部封装的tooltip，downdrop，popup，popover，popconfrim等等需要定位的组件就不用去重写定位了，把定位单独做成一个组件，动画组件是为了in和out都能兼容实现，且频繁触发的时候过渡动画不会跳跃生硬

### 业务代码

```auto
transition和position，两个组件参考vue，model，tooltip，popup，downdrop，popover，popconfrim都能受益
```

### 浏览器

Chrome

### 友好承诺

- [x] 我承诺将本着相互尊重、理解和友善的态度进行交流，共同维护 Layui 良好的社区氛围。

**Comment by loyaoo** [comment]
这样，DatePicker，Dropdown，AutoComplete，Select，Popover，等等很多原生组件显示效果都能受益，显得更细腻，第三方组件也能同样保持细腻感，我是真的非常喜欢用Layui，一直觉得vue那些框架非常繁琐，打包文件非常麻烦，但是他们的细节处理非常细腻，我不是说框架结构，而是UI体验感，非常值得参考

**Comment by loyaoo** [comment]
其他组件可以通过第三方来实现，但是动画组件和定位组件我觉得应该作为核心组件开发，这样第三方也能非常方便的统一调用，进一步使第三方组件的UI更贴合框架整体

**Comment by sentsim** [comment]
1.  不大明白你说的 dropdown 组件的过渡动画具体是指哪一块，是弹出时还是关闭时呢？弹出是有动画过渡的。其他过渡动画可以参考官网示例自主实现：https://layui.dev/docs/2/anim/
2.  dropdown 本身就是基于 `lay.position()` 方法实现的定位，不知道这跟你所说的定位组件有哪些区别。

ref: 
- https://layui.dev/docs/2/dropdown/
- https://github.com/layui/layui/blob/main/src/modules/lay.js#L261-L292
